### PR TITLE
Fix export when not logged in

### LINKED ISSUE #389: App crash on export track
**Describe the bug**
App crashes on Export track .

**To Reproduce**
1. Goto My Tracks Fragment
2.  Select the menu of the track to be exported.
3. Select export track.

**Why app is crashing**
getUser() is null 

**How i will fix**
Check if user logged in or not in case not logged in show snackbar showing message login to export

### PR DESCRIPTION
**Pull request by arvindnegi1**
### Description
getUser() is null so app crash on export .

Fixes #389
#### GIF attached.

- check for user login status before export
- show snackbar messabe in case user not logged in.

### CODE REVIEW

**Comment by arvindnegi1** [comment]
Sir @dewall please review.

**Comment by dewall** [comment]
hi @arvindnegi1, this pull request is heading to the master branch but it should target the develop. We usually do not merge directly to the master branch.

**Comment by arvindnegi1** [comment]
@dewall okay sir i will fix that.

File: org.envirocar.app/src/org/envirocar/app/views/tracklist/TrackListLocalCardFragment.java [hunk 1]
```diff
@@ -167,9 +167,14 @@ public void onUploadTrackClicked(Track track) {
...
-                track.updateMetadata(new TrackMetadata(Util.getVersionString(getActivity()),
-                        mUserManager.getUser().getTermsOfUseVersion()));
-                exportTrack(track);
+                if  (mUserManager.getUser()!=null) {
+                    track.updateMetadata(new TrackMetadata(Util.getVersionString(getActivity()),
+                            mUserManager.getUser().getTermsOfUseVersion()));
+                    exportTrack(track);
+                }
+                else  {
+                    showSnackbar(R.string.trackview_cannot_export_track);
```
Comment (dewall) [inline org.envirocar.app/src/org/envirocar/app/views/tracklist/TrackListLocalCardFragment.java]: I guess you can simply set the terms of use version to null if no user exists. It is okay when not logged in users can export a track.

File: org.envirocar.app/res/values/strings_trackviews.xml [hunk 2]
```diff
@@ -28,5 +28,6 @@
     <string name="trackviews_is_already_uploaded">Track \'%s\' has already been uploaded.</string>
     <string name="trackviews_not_logged_in">Track cannot be uploaded. Please log in first.</string>
     <string name="trackviews_cannot_upload_gps_tracks">GPS Only tracks cannot be uploaded.</string>
+    <string name="trackview_cannot_export_track">Login to export track</string>
```
Comment (dewall) [inline org.envirocar.app/res/values/strings_trackviews.xml]: not required (see below)

**Comment by arvindnegi1** [comment]
@dewall sir, changes Done please review :)

**Pull request merged**